src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala

```diff
@@ -214,6 +214,19 @@ class ResourceService(
       _ <- maybeDeleteResource(resource, samRequestContext)
     } yield ()
 
+  def deletePolicy(policyId: FullyQualifiedPolicyId, samRequestContext: SamRequestContext): IO[Unit] = {
+    for {
+      children <- accessPolicyDAO.listResourceChildren(policyId.resource, samRequestContext)
+      _ <- if (children.nonEmpty) {
+        IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "cannot delete inherited policies")))
```

Comment:
should raise this exception if there are children _and_ the policy applies to any of those children. As written if children exist no policies may be deleted.

Reply:
Oh, I was under the impression that all policies on a resource apply to all of that resource's children. How can we tell if a policy applies to a child?